test/suite/test_prepare_hs04.py

```diff
@@ -0,0 +1,227 @@
+            self.assertEquals(cursor.insert(), 0)
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(1))
+        cursor.close()
+
+        # Call checkpoint
+        self.session.checkpoint()
+
+        cursor = self.session.open_cursor(self.uri)
+        for i in range(1, self.nsessions * self.nkeys):
+            self.session.begin_transaction('isolation=snapshot')
+            key = commit_key + ds.key(self.nrows + i)
+            cursor.set_key(key)
+            self.assertEquals(cursor.remove(), 0)
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(10))
+        cursor.close()
+
+        hs_writes_start = self.get_stat(stat.conn.cache_write_hs)
+        # Have prepared updates in multiple sessions. This should ensure writing
+        # prepared updates to the data store
+        # Insert the same key at timestamp 20, but with prepare updates.
+        sessions = [0] * self.nsessions
+        cursors = [0] * self.nsessions
+        prepare_value = b"ccccc" * 100
+        for j in range (0, self.nsessions):
+            sessions[j] = self.conn.open_session()
+            sessions[j].begin_transaction('isolation=snapshot')
+            cursors[j] = sessions[j].open_cursor(self.uri)
+            # Each session will update many consecutive keys.
+            start = (j * self.nkeys)
+            end = start + self.nkeys
+            for i in range(start, end):
+                cursors[j].set_key(commit_key + ds.key(self.nrows + i))
+                cursors[j].set_value(prepare_value)
+                self.assertEquals(cursors[j].insert(), 0)
+            sessions[j].prepare_transaction('prepare_timestamp=' + timestamp_str(20))
+
+        hs_writes = self.get_stat(stat.conn.cache_write_hs) - hs_writes_start
+
+        # Assert if not writing anything to the history store.
+        self.assertGreaterEqual(hs_writes, 0)
+
+        txn_config = 'read_timestamp=' + timestamp_str(5) + ',ignore_prepare=true'
+        # Search keys with timestamp 5, ignore_prepare=true and expect the cursor search to return 0 (key found)
+        self.search_keys_timestamp_and_ignore(ds, txn_config, commit_value)
+
+        txn_config = 'read_timestamp=' + timestamp_str(20) + ',ignore_prepare=true'
+        # Search keys with timestamp 20, ignore_prepare=true, expect the cursor to return wiredtiger.WT_NOTFOUND
+        self.search_keys_timestamp_and_ignore(ds, txn_config, None)
+
+        prepare_conflict_msg = '/conflict with a prepared update/'
+        txn_config = 'read_timestamp=' + timestamp_str(20) + ',ignore_prepare=false'
+        # Search keys with timestamp 20, ignore_prepare=false and expect the cursor the cursor search to return prepare conflict message
+        self.search_keys_timestamp_and_ignore(ds, txn_config, prepare_conflict_msg, True)
+
+        # If commit is True then commit the transactions else simulate a crash which would
+        # eventualy rollback transactions.
```

Comment:
I think the word "else" is misleading.  We are always going to simulate a crash in this test.

Reply:
Fixed the comment.